pandas/tests/reshape/test_reshape.py

```diff
@@ -298,315 +324,314 @@ def test_just_na(self):
         assert res_series.index.tolist() == [0]
         assert res_series_index.index.tolist() == ['A']
 
-    def test_include_na(self):
+    def test_include_na(self, sparse, dtype):
+        if sparse:
+            pytest.xfail(reason='nan in index is problematic (GH 16894)')
+
+        kwargs = {'sparse': sparse, 'dtype': dtype}
         s = ['a', 'b', np.nan]
-        res = get_dummies(s, sparse=self.sparse)
+        res = get_dummies(s, **kwargs)
         exp = DataFrame({'a': {0: 1, 1: 0, 2: 0},
-                         'b': {0: 0, 1: 1, 2: 0}}, dtype=np.uint8)
+                         'b': {0: 0, 1: 1, 2: 0}}, dtype=dtype)
         assert_frame_equal(res, exp)
 
         # Sparse dataframes do not allow nan labelled columns, see #GH8822
-        res_na = get_dummies(s, dummy_na=True, sparse=self.sparse)
+        res_na = get_dummies(s, dummy_na=True, **kwargs)
```

Comment:
see my comment above, don't use kwargs generally for passing args to test functions, rather pass directly